sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -437,3 +437,288 @@ mod test_kubernetes_bootstrap_token {
+    PidAvailable,
+}
+
+impl TryFrom<&str> for EvictionHardKey {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        serde_plain::from_str::<EvitionSignal>(&input).context(error::InvalidPlainValue {
+            field: "Eviction Hard key",
+        })?;
+        Ok(EvictionHardKey {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(EvictionHardKey, "EvictionHardKey");
+
+#[cfg(test)]
+mod test_kubernetes_eviction_hard_key {
+    use super::EvictionHardKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_eviction_hard_key() {
+        for ok in &[
+            "memory.available",
+            "nodefs.available",
+            "nodefs.inodesFree",
+            "imagefs.available",
+            "imagefs.inodesFree",
+            "pid.available",
+        ] {
+            EvictionHardKey::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_eviction_hard_key() {
+        for err in &["", "storage.available", ".bad", "bad.", &"a".repeat(64)] {
+            EvictionHardKey::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesThresholdValue represents a string that contains a valid kubernetes threshold value.
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct KubernetesThresholdValue {
+    inner: String,
+}
+
+// Regular expression of Kubernetes quantitiy. i.e. 128974848, 129e6, 129M, 123Mi
+lazy_static! {
+    pub(crate) static ref KUBERNETES_QUANTITY: Regex =
+        Regex::new(r"^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$").unwrap();
+}
+
+impl TryFrom<&str> for KubernetesThresholdValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        if input.ends_with("%") {
+            let input_f32 = input[..input.len() - 1]
+                .parse::<f32>()
+                .context(error::InvalidPercentage { input })?;
+            ensure!(
+                (0.0..100.0).contains(&input_f32),
+                error::InvalidThresholdPercentage { input }
+            )
+        } else {
+            ensure!(
+                KUBERNETES_QUANTITY.is_match(input),
+                error::Pattern {
+                    thing: "Kubernetes quantity",
+                    pattern: KUBERNETES_QUANTITY.clone(),
+                    input
+                }
+            );
+        };
+
+        Ok(KubernetesThresholdValue {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(KubernetesThresholdValue, "KubernetesThresholdValue");
+
+#[cfg(test)]
+mod test_kubernetes_threshold_value {
+    use super::KubernetesThresholdValue;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_kubernetes_threshold_value() {
+        for ok in &["10%", "129e6", "10Mi", "1024M", "1Gi"] {
+            KubernetesThresholdValue::try_from(*ok).unwrap();
+        }
+    }
```

Comment:
This doesn't seem to support operators, e.g. `memory.available<100Mi`.

Reply:
Yeah. Shailesh left a pretty similar  comment in my fork PR. And my answer was 
" I do not think It's necessary to use format  `[eviction-signal][operator][quantity]` there. Kubernetes used to have flag `--eviction-hard`, and it takes parameters like this `--eviction-hard=memory.available<500Mi,nodefs.available<1Gi,imagefs.available<100Gi`, and right now this flag has been deprecated, `eviction-hard` parameters should be configured though `kublete-config`, and in `kublete-config`the formate is 
```
apiVersion: kubelet.config.k8s.io/v1beta1
kind: KubeletConfiguration
evictionHard:
    memory.available:  "200Mi"
```
So I think `"memory.available" = "15%"`  is fine here."

What do you think? :)